### ISSUE TITLE
Sprockets 4 require directive from SCSS

### ISSUE THREAD

**Issue by brodock** [rails/sprockets#647]
### Expected behavior

application.css.scss
```
//= random/file
//= another/random/file.css
```

should work as it did in previous versions

### Actual behavior

In both cases, you will get an error like: 

```
Sprockets::FileNotFound: couldn't find file 'random/file' with type 'text/scss'
Checked in these paths: 
```

### System configuration

- Sprockets version 4.0.0
- Ruby version 2.6.3

### Example App (Reproduction) - THIS IS IMPORTANT YOUR ISSUE LIKELY WILL NOT BE RESOLVED WITHOUT THIS

Please provide an [example app](https://www.codetriage.com/example_app) that reproduces the problem. This will save maintainers time so they can spend it fixing your issues instead of trying to build a reproduction case from sparse instructions.

To create an example app, make an application locally that uses your same version of Sprockets. Add files to reproduce the problem. Once you've reproduced the problem add instructions on how to run your example app to a README.md file. Then commit everything to git and push to a new project on github. Once you've pushed your app add a link back to this issue.

[More information about example apps](https://www.codetriage.com/example_app)

**Comment by s-mage** [comment]
same for coffeescript. Problem is here
https://github.com/rails/sprockets/blob/master/lib/sprockets/directive_processor.rb#L215-L217

and here
https://github.com/rails/sprockets/blob/master/lib/sprockets/resolve.rb#L149-L150

As I understood two things are wrong here
* `accept` option should be `nil` when path has extname
* coffee and scss should have the same accepted filetypes as js and css

This issue is hanging for a month here so I'll try to make a PR that fixes this

**Comment by ricardopacheco** [comment]
Same problem here. Waiting PR.

**Comment by s-mage** [comment]
>  I'll try to make a PR that fixes this

The fix is not that easy as I expected, so I shelved it and eventually forgot. I'm sorry. Don't expect me to do anything before the new year.

**Comment by ricardopacheco** [comment]
I tested your PR on my project and it worked correctly. In this case, I'll rollback to the latest version 3.x

**Comment by niborg** [comment]
Any progress on this issue @s-mage ? I suspect this is blocking ability to upgrade Rails for a lot of people. Can you share the specifics of the problem you are running into with the current open PR?

**Comment by s-mage** [comment]
@niborg no, I didn't do anything. I was out of ideas, that's why I asked for help in the PR description. The problem is that my PR is dirty and it breaks a lot of tests. I call it dirty because it tries to use `inverted_transformers` to find the source file and IMO there should be a better way to do it.

**Comment by Josufst** [comment]
Sorry, but what is a PR?

**Comment by MarkerDave** [comment]
I'm having the same problem with Rails 5.0 and sprocket 4.0.2

**Comment by s-mage** [comment]
@MarkerDave 
https://github.com/rails/sprockets/pull/653#issuecomment-828736037 changing `require something.scss` to `require something` helped me

**Comment by Josufst** [comment]
Well,
i didn’t find a solition :-(

krind regards,

Joost van der Linden

Skickad från min iPhöne

**Comment by ikalashniuk** [comment]
I facing this now.
Trying to upgrade Rails from 4.2.0 to 4.2.8.
Renaming *.css to *.scss in project directories and adding copies of gems *.css as *.scss helps.

Now it's time to find a solution for *.js/*.coffee.
